Source/Scene/Cesium3DTileBatchData.js

```diff
@@ -0,0 +1,682 @@
+/*global define*/
```

Comment:
Good job separating this out.  I'm glad it is so reusable.

What do you think is a better suffix than `Data`?  `Textures`?  `Resources`?  `Table`?  This is basically the GPU runtime version of the batch table.

Batch is also subject to rename, but that is TBA.